File: 1-two-sum/1-two-sum.go
func twoSum(nums []int, target int) []int {
    
    // map (number => index)
    m := make(map[int]int)
    
    var inverse int
    for i := range nums {
        inverse = target - nums[i]
        if idx, isFound := m[inverse]; isFound {
            return []int{idx, i}
        }
        m[nums[i]] = i
    }
    return []int{0, 0}
}

File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        i = 0
        j = 0
        
        dic = {}
        for i in range(len(nums)):
            if (target - nums[i]) in dic and dic[target - nums[i]] != i:
                return [i, dic[target - nums[i]]]
            dic[nums[i]] = i
            
                        
        